### PR TITLE
[CELEBORN-1594] Refine dynamicConfig template and prevent NPE

### PR DESCRIPTION
**Pull request by turboFei**
### What changes were proposed in this pull request?

1. seems the quota.yaml is outdated, remove it
2. the config item set in dynamicConfig.yaml.template is not dynamic config, remove the one
3. prevent  NPE when loading dynamicConfig.yaml

### Why are the changes needed?
As title.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Existing UT.

### CODE REVIEW

File: conf/dynamicConfig.yaml.template [hunk 1]
```diff
@@ -16,12 +16,15 @@
 #
 -  level: SYSTEM
    config:
-     celeborn.worker.directMemoryRatioToPauseReceive: 0.75
```
Comment (turboFei) [inline conf/dynamicConfig.yaml.template:19]: seems this config item is not dynamic, I wonder whether it works now

Comment (turboFei) [reply conf/dynamicConfig.yaml.template:19]: image

Currently, there are only 4 dynamic configs for quota management.

File: docs/configuration/index.md [hunk 2]
```diff
@@ -168,7 +168,7 @@ If `celeborn.worker.flusher.buffer.size` is 256 KB, we can have total slots up t
 
 ## Rack Awareness
 
-Celeborn can be rack-aware by setting `celeborn.client.reserveSlots.rackware.enabled` to `true` on client side.
+Celeborn can be rack-aware by setting `celeborn.client.reserveSlots.rackaware.enabled` to `true` on client side.
```
Comment (turboFei) [inline docs/configuration/index.md:171]:
```
  val CLIENT_RESERVE_SLOTS_RACKAWARE_ENABLED: ConfigEntry[Boolean] =
    buildConf("celeborn.client.reserveSlots.rackaware.enabled")
      .withAlternative("celeborn.client.reserveSlots.rackware.enabled")
      .categories("client")
      .version("0.3.1")
      .doc("Whether need to place different replicates on different racks when allocating slots.")
      .booleanConf
      .createWithDefault(false)
```

Comment (turboFei) [reply docs/configuration/index.md:171]: Since 0.3.1, should we use the new config?

File: conf/dynamicConfig.yaml.template (hunk 1 above)
Comment (RexXiong) [reply conf/dynamicConfig.yaml.template:19]: > seems this config item is not dynamic, I wonder whether it works now

yes，seems can be dynamic，but not supported yet.

File: docs/configuration/index.md (hunk 2 above)
Comment (RexXiong) [reply docs/configuration/index.md:171]: We can either keep this or remove it. However, if we decide to remove it, we will need to update the migration guide accordingly.

Comment (turboFei) [reply docs/configuration/index.md]: ok, will keep it

File: conf/dynamicConfig.yaml.template (hunk 1 above)
Comment (turboFei) [reply conf/dynamicConfig.yaml.template:19]: Raised ticket to support more dynamic configs: https://issues.apache.org/jira/browse/CELEBORN-1598

**Review by RexXiong — APPROVED** [review]

**Review by SteNicholas — APPROVED** [review]
LGTM.

**Comment by RexXiong** [comment]
Thanks. merge to main(v0.6.0) and branch-0.5(v0.5.2)

**Pull request closed without merging**